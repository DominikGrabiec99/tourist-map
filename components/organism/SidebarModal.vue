<template>
  <div
    class="absolute top-0 left-0 w-full h-full bg-black-rgba-06"
    @click="closeModal()"
  >
    <div
      class="modal top-0 right-0 absolute h-screen bg-white w-full sm:max-w-md py-4 px-3 z-20"
      @click.stop=""
    >
      <div class="button-container flex justify-end">
        <AtomIcon
          @click="closeModal()"
          class="close-modal bg-transparent text-2xl cursor-pointer text-gray-400"
          :icon="['fas', 'xmark']"
        />
      </div>
      <div
        class="modal-container"
        :class="{
          'modal-container--country':
            modalOptions.name === ESidebarModalName.COUNTRY,
        }"
      >
        <MoleculeCountryModal
          v-if="modalOptions.name === ESidebarModalName.COUNTRY"
          v-bind="modalOptions.props"
          @close-modal="closeModal"
        />
      </div>
    </div>
  </div>
</template>

<script lang="ts" setup>
import EProvideName from '@/ts/enums/ProvideName';
import ESidebarModalName from '@/ts/enums/SidebarModalName';
import IModalOptions from '@/ts/interfaces/ModalOptions';

const modalOptions: IModalOptions = inject(EProvideName.SIDEBAR_MODAL)!;

const closeModal = () => {
  modalOptions.name = ESidebarModalName.NONE;
  if (modalOptions.props?.closeModal) {
    modalOptions.props.closeModal();
  }
  modalOptions.isOpen = false;
  modalOptions.props = {};
};
</script>
