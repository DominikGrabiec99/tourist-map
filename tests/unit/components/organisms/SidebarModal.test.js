import { describe, it } from 'vitest';
import { shallowMount } from '@vue/test-utils';

import SidebarModal from '@/components/organism/SidebarModal.vue';

import ESidebarModalName from '@/ts/enums/SidebarModalName';

describe('SidebarModal.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(SidebarModal, {
      global: {
        provide: {
          SIDEBAR_MODAl: {
            name: ESidebarModalName.COUNTRY,
            isOpen: true,
            props: {
              closeModal: () => {},
            },
          },
        },
      },
    });
  });

  it('render component', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should close modal', () => {
    expect(wrapper.vm.modalOptions.isOpen).toBeTruthy();
    wrapper.vm.closeModal();
    expect(wrapper.vm.modalOptions.isOpen).toBeFalsy();
  });

  it('handle close modal button', () => {
    wrapper.find('.close-modal').trigger('click');
    expect(wrapper.vm.modalOptions.isOpen).toBeFalsy();
  });

  it('handle close modal by wrapper', () => {
    wrapper.trigger('click');
    expect(wrapper.vm.modalOptions.isOpen).toBeFalsy();
  });
});
