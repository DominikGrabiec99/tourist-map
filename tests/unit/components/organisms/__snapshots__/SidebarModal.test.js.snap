// Vitest Snapshot v1

exports[`SidebarModal.vue > render component 1`] = `
<div
  class="absolute top-0 z-50 left-0 w-full h-full bg-black-rgba-06 dark:bg-white-rgba-06"
>
  <div
    class="modal top-0 right-0 absolute h-screen bg-white dark:bg-neutral-900 w-full sm:max-w-md py-4 px-3 z-20"
  >
    <div
      class="button-container flex justify-end"
    >
      <atomicon
        class="close-modal bg-transparent text-2xl cursor-pointer text-gray-400"
        icon="fas,xmark"
      />
    </div>
    <div
      class="modal-container modal-container--country"
    >
      <moleculecountrymodal
        closemodal="() => {}"
        data-test="countryModal"
      />
    </div>
  </div>
</div>
`;
